src/python/pants/engine/subsystem/native.py

```diff
@@ -166,7 +168,7 @@
     ExecutionStat execution_execute(RawScheduler*);
     RawNodes* execution_roots(RawScheduler*);
 
-    void validator_run(RawScheduler*, TypeId*, uint64_t);
+    Value* validator_run(RawScheduler*, TypeId*, uint64_t);
```

Comment:
I think the answer is approximately: call `context.drop_handles` for the Value... but I don't know if we want to worry about it since this API is very low traffic. Maybe just leave a TODO, and if we end up with enough of those we can look for a clean way to make it easy to do the right thing.